### DISCUSSION TITLE
socket hang up after 5000ms

### DISCUSSION THREAD

**Discussion by indolent-developer** [sindresorhus/got#2119] [Q&A] [answered]
Hi I am using got library and facing some issue. We are getting socket hangup after a time of 500ms. Do we know what can cause this behavior. The application logs show request leaving client service but the server service doesn't show any request received.

Can got have a timeout where it resets connection automatically? 

some sample code 

const { body, statusCode, headers, } = await got(url, {
    headers: {
      'Keep-Alive': 'timeout=10, max=100',
      ...requestHeaders,
    },
    
    got version . "got-cjs": "^12.0.1"

**Comment by sindresorhus** [comment, marked as answer]
"Socket hang up" errors typically occur when the connection is closed unexpectedly, often due to timeouts or network issues. Got doesn't have a hardcoded 5000ms timeout - the default socket timeout is actually undefined (no timeout).

Here are some things to check.

Configure explicit timeuts:

```javascript
import got from 'got';

await got('https://api.example.com', {
	timeout: {
		socket: 10000,    // 10 seconds of inactivity
		response: 30000   // 30 seconds to receive response
	},
	retry: {
		limit: 2
	}
});
```

Keep-alive config:

If you're using custom agents, make sure keep-alive is configured properly:

```javascript
import https from 'https';
import got from 'got';

const agent = new https.Agent({
	keepAlive: true,
	keepAliveMsecs: 30000,
	maxSockets: 50,
	timeout: 10000
});

await got('https://api.example.com', {
	agent: {
		https: agent
	}
});
```

some common causes:
- The server is taking longer than 5 seconds to respond (could be a load balancer or proxy timeout)
- Network infrastructure between client and server has a timeout
- The server closes idle connections after 5 seconds

Try adding more detailed error logging to see where the request fails:

```javascript
try {
	const response = await got('https://api.example.com');
} catch (error) {
	console.log('Error code:', error.code);
	console.log('Timings:', error.timings);
	console.log('Request options:', error.options);
}
```

The `timings` object will show you exactly where the delay occured.
